src/sha.h

```diff
@@ -5,28 +5,11 @@
 #include "autoconfig.h"
 #endif
 
-#if HAVE_LIBCRYPTO || HAVE_COMMONCRYPTO
+#if HAVE_LIBCRYPTO
 #include <openssl/opensslv.h>
 
 #include "arch.h"
-#if HAVE_COMMONCRYPTO || (!AC_BUILT &&	  \
-	!defined(SIMD_COEF_32) && defined(__APPLE__) && defined(__MACH__) && \
-	 defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && \
-	 __MAC_OS_X_VERSION_MIN_REQUIRED >= 1070)
-/* Mitigate CommonCrypto name clashes */
-#include "md4.h"
-#include "md5.h"
-#define COMMON_DIGEST_FOR_OPENSSL 1
-#include <CommonCrypto/CommonDigest.h>
-#ifndef SHA_CBLOCK
-#define SHA_CBLOCK CC_SHA1_BLOCK_BYTES
-#endif
-#ifndef SHA_LBLOCK
-#define SHA_LBLOCK CC_SHA1_BLOCK_LONG
-#endif
-#else
 #include <openssl/sha.h>
-#endif
```

Comment:
After this removal, we could avoid mixing the order of including OpenSSL vs. our own headers here.

Reply:
I guess we can actually drop the `#include "arch.h"` now... it was needed for checking `SIMD_COEF_32`.

Reply:
Or we could include "arch.h" in top, which in turn will source autoconfig.h if applicable

Reply:
This make it more like eg. md4.h
```diff
diff --git a/src/sha.h b/src/sha.h
index 244fd46ef..cadf281d4 100644
--- a/src/sha.h
+++ b/src/sha.h
@@ -1,14 +1,10 @@
 #ifndef JOHN_SHA_H
 #define JOHN_SHA_H
 
-#if AC_BUILT
-#include "autoconfig.h"
-#endif
+#include "arch.h" /* also includes autoconfig.h for HAVE_LIBCRYPTO */
 
 #if HAVE_LIBCRYPTO
 #include <openssl/opensslv.h>
-
-#include "arch.h"
 #include <openssl/sha.h>
 
 /* For the abuse in pbkdf2_hmac_sha1.h and mscash2_fmt_plug.c */
```

Reply:
Hm, no, while md4.h may need stuff from arch.h, we do not here. I'll just drop the arch.h